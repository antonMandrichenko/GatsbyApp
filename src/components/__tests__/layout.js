import React from 'react'
import { render, fireEvent, waitForElement } from '@testing-library/react'
// import axiosMock from 'axios'
import Layout from '../layout'

test('loads and displays greeting', async () => {
  const url = '/greeting'
  const { getByText, getByRole, asFragment } = render(<Layout />)

//   axiosMock.get.mockResolvedValueOnce({
//     data: { greeting: 'hello there' },
//   })

//   fireEvent.click(getByText('Load Greeting'))

//   const greetingTextNode = await waitForElement(() => getByRole('heading'))

//   expect(axiosMock.get).toHaveBeenCalledTimes(1)
//   expect(axiosMock.get).toHaveBeenCalledWith(url)
//   expect(getByRole('heading')).toHaveTextContent('hello there')
//   expect(getByRole('button')).toHaveAttribute('disabled')
  expect(asFragment()).toMatchSnapshot()
})