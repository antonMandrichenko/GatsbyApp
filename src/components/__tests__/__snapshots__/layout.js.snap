// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads and displays greeting 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6"
      >
        <a
          class="makeStyles-title-3"
          href="/"
        >
          Tickets app
        </a>
      </h6>
    </div>
  </header>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
  >
    <div
      class="MuiGrid-root makeStyles-margin-4 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-10 MuiGrid-grid-md-8 MuiGrid-grid-lg-6"
    />
  </div>
</DocumentFragment>
`;
